experimental/web/sample_webgpu/main.c

```diff
@@ -758,8 +765,8 @@ static iree_status_t process_call_outputs(
         .type = IREE_HAL_TRANSFER_COMMAND_TYPE_COPY,
         .copy =
             {
-                .source_buffer = source_buffer,
-                .source_offset = data_offset,
+                .source_buffer = source_buffer_allocated,
+                .source_offset = source_offset,
```

Comment:
Some of this async / loop-based readback code could be moved to a shared place with the rest of the C runtime code, where it would be easier to test.